src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -473,6 +533,27 @@ public ReadOnlySpan<byte> GetNullTerminatedBytes()
 
         #endregion
 
+        #region Dispose
+
+        public void Dispose()
+        {
+            if (_disposed)
+                return;
+
+            _timeoutCts.Dispose();
+
+            _disposed = true;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        void CheckDisposed()
```

Comment:
While it's correct and good practice to check disposed, I'm not sure it's needed in this case. For one thing, this method is only being called in two places (Ensure, AllocateOversize) - to be really correct it needs to be called from every public method of this type (e.g. ReadInt32) - but I wouldn't necessarily want to add all those checks everywhere all the time.

More importantly this is essentially an internal type, although it's public so that external type handler plugins can be written. The responsibility for making sure nothing is called on this after disposal is in Npgsql's core - if that ever happens, that's a bug in Npgsql we have to fix. So I'd consider removing this entirely.